dev/benchmarks/macrobenchmarks/lib/src/web/recorder.dart

```diff
@@ -1182,3 +1188,76 @@ void _dispatchEngineBenchmarkValue(String name, double value) {
     listener(value);
   }
 }
+
+abstract class EmptyRecorder extends Recorder implements FrameRecorder {
```

Comment:
This recorder does not look empty. It looks like `WidgetRecorder` without the `drawFrameDuration` timeline. If that's all the difference, why not use `WidgetRecorder` and simply ignore the `drawFrameDuration` numbers?

If this class is really necessary, let's discuss how it fits with the other recorder classes. Perhaps it should be a super-class of `WidgetRecorder`? We should also add substantially more documentation for this class. In particular we should document when it should be used.